Update Grafana, Add Jaeger to visualize data in Promscale

VineethReddy02 opened the pull request:
# Description

Add Jaeger to Visualize data in Promscale 

# Version

Which documentation version does this PR apply to?

- [x] Latest (Default)

# Links

Fixes https://github.com/timescale/docs/issues/923
          https://github.com/timescale/docs/issues/924

Recreated a new PR pointing to `tracing-ga` branch, previous this was #961

Loquacity commented on promscale/page-index/page-index.js:
```diff
@@ -145,9 +145,25 @@ module.exports = [
+            href: 'grafana',
+            tags: ['promscale', 'configure', 'prometheus', 'metrics'],
+            keywords: ['Promscale', 'Grafana'],
+            excerpt: 'Grafana to visualise data in Promscale',
```
```suggestion
            excerpt: 'Grafana to visualize data in Promscale',
```

Loquacity commented on promscale/visualize-data/grafana.md:
```diff
@@ -0,0 +1,67 @@
+
+To connect Grafana with Promscale, you should have Grafana running. If you do not have Grafana installed, follow the [Grafana installation docs][grafana-install].
+
+This section shows you how to connect Promscale as a Prometheus, Jaeger and PostgreSQL data sources in
```
```suggestion
This section shows you how to connect Promscale to Prometheus, Jaeger, and PostgreSQL data sources in
```

Loquacity commented on promscale/visualize-data/grafana.md:
```diff
@@ -0,0 +1,67 @@
+
+<procedure>
+
+## Promscale as Prometheus datasource
```
Missing a verb here. "Connecting"?

VineethReddy02 replied:
Added it as "Configuring"

Loquacity commented on promscale/visualize-data/grafana.md:
```diff
@@ -0,0 +1,67 @@
+
+</procedure>
+
+Once you have configured Promscale as a Prometheus data source in Grafana, you can create panels that are populated with data using PromQL
```
```suggestion
When you have configured Promscale as a Prometheus data source in Grafana, you can create panels that are populated with data using PromQL
```

Loquacity commented on promscale/visualize-data/grafana.md:
```diff
@@ -0,0 +1,67 @@
+
+<procedure>
+
+## Promscale as Jaeger datasource
```
Missing a verb

Loquacity commented on promscale/visualize-data/grafana.md:
```diff
@@ -0,0 +1,67 @@
+
+</procedure>
+
+You can now filter and view traces stored in Promscale using Grafana. To visualize your traces, go to the “Explore” section of Grafana. You will be taken to the traces filtering panel.
```
```suggestion
You can now filter and view traces stored in Promscale using Grafana. To visualize your traces, go to the `Explore` section of Grafana to see the traces filtering panel.
```

Loquacity commented on promscale/visualize-data/grafana.md:
```diff
@@ -0,0 +1,67 @@
+
+<procedure>
+
+## Promscale as PostgreSQL datasource
```
Missing verb

Loquacity commented on promscale/visualize-data/grafana.md:
```diff
@@ -0,0 +1,67 @@
+1.  Configure the data source settings:
+    *   In the `Name` field, type `Promscale-SQL`.
+    *   In the Host field, type `<host>:<port>`, where host and port need to be 
+        obtained from the service url you copied when you created the Timescale cloud service. The format of that url is `postgresql://[user[:password]@][host][:port][/dbname][?param1=value1&...]`
```
```suggestion
        obtained from the service URL you copied when you created the Timescale cloud service. The format of that URL is `postgresql://[user[:password]@][host][:port][/dbname][?param1=value1&...]`
```

Loquacity commented on promscale/visualize-data/grafana.md:
```diff
@@ -0,0 +1,67 @@
+    *   In the `Name` field, type `Promscale-SQL`.
+    *   In the Host field, type `<host>:<port>`, where host and port need to be 
+        obtained from the service url you copied when you created the Timescale cloud service. The format of that url is `postgresql://[user[:password]@][host][:port][/dbname][?param1=value1&...]`
+    *   In the `Database` field, type the dbname from the service url.
```
```suggestion
    *   In the `Database` field, type the database name from the service URL.
```

Loquacity commented on promscale/visualize-data/grafana.md:
```diff
@@ -0,0 +1,67 @@
+    *   In the Host field, type `<host>:<port>`, where host and port need to be 
+        obtained from the service url you copied when you created the Timescale cloud service. The format of that url is `postgresql://[user[:password]@][host][:port][/dbname][?param1=value1&...]`
+    *   In the `Database` field, type the dbname from the service url.
+    *   In the User and Password fields, type the user and password from the service.
```
```suggestion
    *   In the `User` and `Password` fields, type the user and password of the service.
```

Loquacity commented on promscale/visualize-data/grafana.md:
```diff
@@ -0,0 +1,67 @@
+        obtained from the service url you copied when you created the Timescale cloud service. The format of that url is `postgresql://[user[:password]@][host][:port][/dbname][?param1=value1&...]`
+    *   In the `Database` field, type the dbname from the service url.
+    *   In the User and Password fields, type the user and password from the service.
+    *   Change the `TLS/SSL Mode` to require as the service url by default contains 
+        the TLS mode as required.
```
I can't work out what this sentence is trying to say.

Loquacity commented on promscale/visualize-data/grafana.md:
```diff
@@ -0,0 +1,67 @@
+    *   In the User and Password fields, type the user and password from the service.
+    *   Change the `TLS/SSL Mode` to require as the service url by default contains 
+        the TLS mode as required.
+    *   Change the `TLS/SSL Method` File system path.
```
```suggestion
    *   Change the `TLS/SSL Method` file system path.
```

Loquacity commented on promscale/visualize-data/grafana.md:
```diff
@@ -0,0 +1,67 @@
+        the TLS mode as required.
+    *   Change the `TLS/SSL Method` File system path.
+    *   Use the default values for all other settings.
+    *   In the PostgreSQL details section enable the TimescaleDB option.
```
```suggestion
    *   In the PostgreSQL details section, check the `TimescaleDB` option.
```

VineethReddy02 replied:
I prefer `enable` here as the enabling TimescaleDB option isn't a checkbox it's a kind of bar icon that needs to be clicked to enable. So the word enable represents the user that it needs to be turned on.

Loquacity commented on promscale/visualize-data/grafana.md:
```diff
@@ -0,0 +1,67 @@
+
+</procedure>
+
+You can now create panels that use Promscale as a PostgreSQL data source, using SQL queries to feed the chart
```
```suggestion
You can now create panels that use Promscale as a PostgreSQL data source, using SQL queries to feed the chart.
```

Loquacity commented on promscale/visualize-data/jaeger.md:
```diff
@@ -0,0 +1,67 @@
+# Visualise Promscale traces in Jaeger
```
```suggestion
# Visualize Promscale traces in Jaeger
```

Loquacity commented on promscale/visualize-data/jaeger.md:
```diff
@@ -0,0 +1,67 @@
+This section shows you how to integrate
+[Jaeger][jaeger-ui] with Promscale.
+
+To visualize traces in Jaeger with Promscale, you need to have Jaeger query component running that queries and visualizes traces from Promscale. 
```
```suggestion
To visualize traces in Jaeger with Promscale, you need to have the Jaeger query component running. The Jaeger component queries and visualizes traces from Promscale. 
```

Loquacity commented on promscale/visualize-data/jaeger.md:
```diff
@@ -0,0 +1,67 @@
+
+<procedure>
+
+## Connect Jaeger query with Promscale
```
```suggestion
## Connecting a Jaeger query with Promscale
```

Loquacity commented on promscale/visualize-data/jaeger.md:
```diff
@@ -0,0 +1,67 @@
+1.  Deploy the Jaeger Query component from the Jaeger 
+    [deployments page][jaeger-deployments], if you aren't already running the Jaeger query.
+    
+    **Note**: For Jaeger to visualize traces in Promscale. We need Jaeger Query of version `1.30` or above. 
```
Move this to above the procedure, with the "Before you begin, make sure..." phrasing. Also remove the first-person pronoun.

Loquacity commented on promscale/visualize-data/jaeger.md:
```diff
@@ -0,0 +1,67 @@
+    
+    **Note**: For Jaeger to visualize traces in Promscale. We need Jaeger Query of version `1.30` or above. 
+    
+1.  Set the Jaeger query configuration through environment variables 
```
Can we split this up to make it clearer?

VineethReddy02 replied:
The action described in this step is setting up the environment variable. I do not like/prefer breaking the same action into multiple steps.

Loquacity commented on promscale/visualize-data/jaeger.md:
```diff
@@ -0,0 +1,67 @@
+
+</procedure>
+
+Visualising traces from Promscale in Jaeger query using the filters offered on the left search menu
```
```suggestion
Visualizing traces from Promscale in Jaeger query using the filters offered on the left search menu
```

This sentence is incomplete. What are you trying to say here?

VineethReddy02 replied:
This is a short description for an image after this phrase.

Loquacity commented on promscale/visualize-data/jaeger.md:
```diff
@@ -0,0 +1,67 @@
+
+<img class="main-content__illustration" src="https://s3.amazonaws.com/assets.timescale.com/images/misc/jaeger-homepage-query-results.png" alt="Sample output for Jaeger query results"/>
+
+## Using Docker for setting up Jaeger Query with Promscale
```
Only use gerunds for procedure headings.

Loquacity replied:
```suggestion
## Use Docker for setting up Jaeger Query with Promscale
```

Rajakavitha1 replied:
Hi @Loquacity,  considering that we need to use gerunds for procedure headings, how about: Setting up Jaeger Query with Promscale

Loquacity replied:
This isn't a procedure, though?

Rajakavitha1 replied:
> This isn't a procedure, though?

Right!!! Thanks makes sense.

Loquacity commented on promscale/visualize-data/jaeger.md:
```diff
@@ -0,0 +1,67 @@
+
+<procedure>
+
+### Connecting Promscale and Jaeger Query
```
```suggestion
### Connecting Promscale and a Jaeger query
```

Rajakavitha1 commented:
Installed and validated the steps to configure Grafana
![Uploading Screenshot 2022-03-28 at 4.53.30 PM.png…]()

Rajakavitha1 reviewed (commented):
Validated and updated the steps for Grafana. Need to install Jaeger to validate the steps in Jaeger.md

Loquacity commented:
Why have we gone back to bolding instead of backticks?

Rajakavitha1 commented:
> Why have we gone back to bolding instead of backticks?

I am sorry, my bad I am so used to the Google Developer guide. I forgot that we were not following that yet. I will revert the changes.

Rajakavitha1 commented:
Hi @VineethReddy02 !!! 

Just wanted to check if you got an opportunity to incorporate the review feedback that @ramonguiu  had provided about instructions being specific to docker in the PR[961](https://github.com/timescale/docs/pull/961) in this PR

Rajakavitha1 commented:
Hi @VineethReddy02 !!! 

Just wanted to check if you got an opportunity to incorporate the review feedback that @ramonguiu  had provided about instructions being specific to docker in the PR[961](https://github.com/timescale/docs/pull/961) in this PR

VineethReddy02 commented:
@Rajakavitha1 You, The comments added by @ramonguiu on instructions being docker focused are removed and added generic instructions that would match with all types of environments, however as an example we added docker way of configuring at the end as heads-up info.

VineethReddy02 commented on promscale/visualize-data/grafana.md:
```diff
@@ -0,0 +1,80 @@
+
+## Configuring Promscale as PostgreSQL data source
+
+To configure Promscale as PostgreSQL data source you need details such as host, port, database, user, and password. These are available in the service URL that you copied when you created the Timescale cloud service. The format of that URL is `postgresql://[user[:password]@][host][:port][/dbname][?param1=value1&...]`
```
@Rajakavitha1 here the context isn't **Timescale Cloud Service** the reference blog post that we published is focused on Timescale cloud service being configured as the data source in Grafana. I would change the phrase from
```
These are available in the service URL that you copied when you created the Timescale cloud service. The format of that URL is
```

to


```
These details should be available in the PostgreSQL service URL. The format of that URL should be
```

VineethReddy02 reviewed (commented):
Overall, LGTM

Rajakavitha1 commented:
Hi @VineethReddy02 !!! I just pushed changes to the Jaegar content. Please review the changes and let me know if it is fine.

Rajakavitha1 commented:
I have validated the content and it is LGTM from me.

Loquacity commented on promscale/visualize-data/grafana.md:
```diff
@@ -0,0 +1,106 @@
+# Connect Promscale and Grafana
+
+Grafana ships with a built-in Prometheus, PostgreSQL, and other data source
+plugins that allows you to query and visualize data from a compatible database.
+To Add a data source in Grafana you need to be a user with oraganization admin
+role previleges.
```
```suggestion
Grafana ships with built-in Prometheus, PostgreSQL, and other data source
plugins that allow you to query and visualize data from a compatible database.
To add a data source in Grafana you must be signed as a user with organization 
adminatration role privileges.
```

Loquacity commented on promscale/visualize-data/grafana.md:
```diff
@@ -0,0 +1,106 @@
+role previleges.
+
+To connect Grafana with Promscale, install and run Grafana version 5.3 or later.
+For information about installing Grafana, see [Grafana installation
+docs][grafana-install].
```
```suggestion
For information about installing Grafana, see the
[Grafana installation documentation][grafana-install].
```

Loquacity commented on promscale/visualize-data/grafana.md:
```diff
@@ -0,0 +1,106 @@
+
+## Configuring Promscale as Jaeger data source
+
+Grafana ships with built-in support for Jaeger, which provides open source,
+end-to-end distributed tracing.
```
This goes above the procedure.

Loquacity commented on promscale/visualize-data/grafana.md:
```diff
@@ -0,0 +1,106 @@
+
+## Configuring Promscale as PostgreSQL data source
+
+To configure Promscale as PostgreSQL data source you need details such as host,
+port, database, user, and password. These are available in the PostgreSQL service. The format of that URL is `postgresql://[user[:password]@][host][:port][/dbname][?param1=value1&...]`
```
This goes above the procedure

Loquacity commented on promscale/visualize-data/jaeger.md:
```diff
@@ -0,0 +1,69 @@
+# Visualize Promscale traces in Jaeger
+
+Jaeger, is an open source distributed tracing system used for monitoring and troubleshooting microservices-based distributed systems.
+
+This section shows you how to integrate [Jaeger][jaeger-ui] with Promscale.
+
+To visualize traces in Jaeger with Promscale, you need to have the Jaeger component `jaeger-query` running. This Jaeger component queries and visualizes traces from Promscale. 
+
+Before you begin, deploy the component `jaeger-query` from the Jaeger [deploymentspage][jaeger-deployments]. Ensure that the version of the component is 1.30 or later. 
```
```suggestion
Jaeger is an open source distributed tracing system used for monitoring and 
troubleshooting microservices-based distributed systems. To visualize traces 
in Jaeger with Promscale, you need to have the `jaeger-query` component 
running. This Jaeger component queries and visualizes traces from Promscale. 

This section shows you how to integrate [Jaeger][jaeger-ui] with Promscale.

Before you begin, deploy the jaeger-query` component from the Jaeger 
[deployments page][jaeger-deployments]. Ensure that you are using `jaeger-query`
version 1.30 or later. 
```

Loquacity commented on promscale/visualize-data/grafana.md:
```diff
@@ -1,9 +1,8 @@
+Grafana ships with built-in Prometheus, PostgreSQL, and other data source
+plugins that allow you to query and visualize data from a compatible database.
+To add a data source in Grafana you must be signed as a user with organization 
+adminatration role privileges.
```
```suggestion
administration role privileges.
```

Loquacity commented on promscale/visualize-data/grafana.md:
```diff
@@ -0,0 +1,104 @@
+
+Grafana ships with built-in support for Jaeger, which provides open source,
+end-to-end distributed tracing.
+<procedure>
```
```suggestion

<procedure>
```

ramonguiu commented on promscale/visualize-data/grafana.md:
```diff
@@ -0,0 +1,105 @@
+1.  Click `Add data source`to see a list of all supported data sources.
+1.  Type `Prometheus` in the search field and click `Select`.
+1.  Configure the data source:
+    *   In the `Name`field, type `Promscale-metrics`.
```
```suggestion
    *   In the `Name`field, type `Promscale-PromQL`.
```

You can also query metrics with SQL. This is about PromQL and I think it's better to name it like this but I don't have strong opinions here.

ramonguiu commented on promscale/visualize-data/grafana.md:
```diff
@@ -0,0 +1,105 @@
+1.  Click `Add data source`to see a list of all supported data sources.
+1.  Type `Jaeger` in the search field and click `Select`.
+1.  Configure the data source settings:
+    *   In the `Name`field, type `Promscale-traces`.
```
```suggestion
    *   In the `Name`field, type `Promscale-Jaeger`.
```

You can also query traces with SQL. This is about querying via the Jaeger api and I think it's better to name it like this but I don't have strong opinions here.

ramonguiu commented on promscale/visualize-data/grafana.md:
```diff
@@ -0,0 +1,105 @@
+<procedure>
+
+To configure Promscale as PostgreSQL data source you need details such as host,
+port, database, user, and password. These are available in the PostgreSQL service. The format of that URL is `postgresql://[user[:password]@][host][:port][/dbname][?param1=value1&...]`
```
```suggestion
port, database, user, and password of the underlying TimescaleDB / PostgreSQL database used by Promscale.
```
There is no such thing as PostgreSQL service (service is a Timescale Cloud name) and the parameters are when you start the database are not passed like in that URL.I think we are listing the url because this is how we show it in Timescale Cloud and we had it in our blog. I think we can just remove this whole sentence.

ramonguiu commented on promscale/visualize-data/grafana.md:
```diff
@@ -0,0 +1,105 @@
+src="https://s3.amazonaws.com/assets.timescale.com/images/misc/getting-started-with-promscale-grafana-dashboard.png"
+alt="Sample output for PromQl query"/>
+
+Grafana ships with built-in support for Jaeger, which provides open source,
```
Shouldn't this be part of the Configuring Promscale as Jaeger data source section? I understand we use a procedure but it's weird to see this text directly after talking about the Prometheus data source with not break in between (some title usually).

ramonguiu commented on promscale/visualize-data/grafana.md:
```diff
@@ -0,0 +1,105 @@
+</procedure>
+
+You can now filter and view traces stored in Promscale using Grafana. To
+visualize your traces, click the **Explore** button to see the traces filtering
```
```suggestion
visualize your traces, click the **Explore** button and select the `Promscale-traces` data source to see the traces filtering
```

Replace Promscale-traces with Promscale-Jaeger if you replace the name of the data source above.

Rajakavitha1 replied:
Hi @ramonguiu ! When I validated the steps I noticed that there was an option to explore at the bottom left when you are configuring the data source. Considering that the user just configured the Jaeger data source and has still not exited the page they may not have to select the `Promscale-traces` ![Screenshot 2022-04-14 at 4 56 19 PM](https://user-images.githubusercontent.com/37059749/163391903-41944d1d-c5d4-4807-9cb2-361a8a422c6d.png)

ramonguiu replied:
Ah! Sounds great!

ramonguiu commented on promscale/visualize-data/grafana.md:
```diff
@@ -0,0 +1,105 @@
+
+<procedure>
+
+To configure Promscale as PostgreSQL data source you need details such as host,
```
Again we transition to setting up a PostgreSQL data source without and section break (a title for example).

ramonguiu commented on promscale/visualize-data/grafana.md:
```diff
@@ -0,0 +1,105 @@
+1.  Configure the data source settings:
+    *   In the `Name`field, type `Promscale-SQL`.
+    *   In the `Host`field, type the IP address or hostname and optional port of
+        your PostgreSQL instance. 
```
```suggestion
        your TimescaleDB / PostgreSQL instance. 
```

ramonguiu commented on promscale/visualize-data/grafana.md:
```diff
@@ -0,0 +1,105 @@
+    *   In the `Name`field, type `Promscale-SQL`.
+    *   In the `Host`field, type the IP address or hostname and optional port of
+        your PostgreSQL instance. 
+    *   In the `Database` field, type the name of the PostgreSQL database.
```
@VineethReddy02 don't we know the name of the database? Isn't it tsdb?

ramonguiu commented on promscale/visualize-data/grafana.md:
```diff
@@ -0,0 +1,105 @@
+        your PostgreSQL instance. 
+    *   In the `Database` field, type the name of the PostgreSQL database.
+    *   In the `User`and `Password`fields, type the `user name` and the
+        `password for the database`.
```
```suggestion
        `password` for the database.
```

ramonguiu commented on promscale/visualize-data/grafana.md:
```diff
@@ -0,0 +1,105 @@
+    *   In the `Host`field, type the IP address or hostname and optional port of
+        your PostgreSQL instance. 
+    *   In the `Database` field, type the name of the PostgreSQL database.
+    *   In the `User`and `Password`fields, type the `user name` and the
```
```suggestion
    *   In the `User`and `Password` fields, type the `user name` and the
```

ramonguiu commented on promscale/visualize-data/jaeger.md:
```diff
@@ -0,0 +1,71 @@
+
+This section shows you how to integrate [Jaeger][jaeger-ui] with Promscale.
+
+Before you begin, deploy the jaeger-query` component from the Jaeger 
```
```suggestion
Before you begin, deploy the `jaeger-query` component from the Jaeger 
```

ramonguiu commented on promscale/visualize-data/jaeger.md:
````diff
@@ -0,0 +1,71 @@
+    ```
+    where: `<PROMSCALE_HOST>:9202` is the Promscale gRPC endpoint.
+
+1.  Start the `jaegar-query` component, it should be successfully connected with
````
```suggestion
1.  Start the `jaeger-query` component, it should be successfully connected with
```

ramonguiu commented on promscale/visualize-data/jaeger.md:
```diff
@@ -0,0 +1,71 @@
+
+<procedure>
+
+## Connecting a Jaeger query with Promscale
```
```suggestion
## Connecting Jaeger Query with Promscale
```

ramonguiu commented on promscale/visualize-data/jaeger.md:
```diff
@@ -0,0 +1,71 @@
+<procedure>
+
+### Connecting Promscale and a Jaeger query
+1.  Install Jaeger from the [official Docker image][jaeger-docker]:
```
```suggestion
1.  Run Jaeger query using the [official Docker image][jaeger-docker]:
```
